dist/tools/licenses/check_packages.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: Copyright (C) 2025 ML!PA Consulting GmbH
+
+# Checks all packages' Makefiles for SPDX compatibility of their license
+# declarations.
+
+cd "$(dirname \"$(readlink -f \"$0\")\")"
```

Comment:
```suggestion
cd "$(dirname "$(readlink -f "$0")")"
```

Although counter-intuititve, the quotes inside of quotes shouldn't be escaped: https://www.shellcheck.net/wiki/SC2086

Note that $( ) starts a new context, and variables in it have to be quoted independently:
```sh
echo "This $variable is quoted $(but this $variable is not)"
echo "This $variable is quoted $(and now this "$variable" is too)"
```

Reply:
Thanks for the input on this.